hicommonwealth/commonwealth
# Intelligent Message Routing

## Link to Issue
Closes: #8369 

## Description of Changes
- Moves all sub-level schemas out of `events.schemas.ts` since we export all the schemas as a type.
- Added the `getRoutingKey` function to the Broker interface/port.
- Added simplified/updated RabbitMQ/Rascal configs to use the new routing keys.

## Test Plan
- Start RabbitMQ in Docker `pnpm start-rmq`
- Start the API
- Start the message relayer `pnpm start-message-relayer`
- Go to a community that does not have contests
- Create a thread
  - The message relayer should logged published
- Go to http://localhost:15672/ and sign in with username 'guest' and password 'guest'
- Go to 'queues'
  - There should be no pending messages in the `ContestWorkerPolicyQueue`
- Now go to a community with contests enabled or a create a new one
- Create a thread in that community
- Go back to the RabbitMQ dashboard and the `ContestWorkerPolicyQueue` should have a single message.

## Deployment Plan

1. Delete the old `ThreadCreated` and `ThreadUpvoted` bindings from the `ContestWorkerPolicyQueue` from the RabbitMQ dashboard _**after deployment**_.

## Other Considerations

-
